jobs/jobs/utils.py

```diff
@@ -585,7 +590,147 @@ async def get_job_progress(
         )
 
     response.update({"mode": str(job.mode)})
-    return response  # type: ignore
+    if job.mode == JobMode.Manual:
+        await update_manual_job_status(session, job, response)
+        return response
+
+    elif job.mode == JobMode.Automatic:
+        return await handle_automatic_job(
+            session, job, response, job_id, api_client
+        )
+
+    logger.warning(
+        f"Unknown job mode: {job.mode}, treating as manual by default."
+    )
+    await update_manual_job_status(session, job, response)
+    return response
+
+
+async def update_manual_job_status(
+    session: Session, job: CombinedJob, response: Dict[str, int]
+) -> None:
+    """Updates the job status based on the DAG execution response."""
+    if response.get("finished", 0) > 0:
+        if response["finished"] == response["total"]:
+            db_service.update_job_status(session, job, Status.finished)
+        else:
+            db_service.update_job_status(session, job, Status.in_progress)
+
+
+async def handle_automatic_job(
+    session: Session,
+    job: CombinedJob,
+    response: Dict[str, int],
+    job_id: int,
+    api_client: client.ApiClient,
+) -> Dict[str, int]:
+    """Handles automatic jobs with or without a pipeline."""
+    if response.get("total", 0) == 0:
+        return await handle_pipeline_driven_job(
+            session, job, response, job_id, api_client
+        )
+    else:
+        await update_manual_job_status(session, job, response)
+        return response
+
+
+async def handle_pipeline_driven_job(
+    session: Session,
+    job: CombinedJob,
+    response: Dict[str, int],
+    job_id: int,
+    api_client: client.ApiClient,
+) -> Dict[str, int]:
+    """Handles automatic jobs that rely on pipeline DAG execution."""
+    pipeline_id = job.pipeline_id
+    dag_run_id = f"{pipeline_id}:{job_id}"
+
+    await activate_pipline(pipeline_id, api_client)
+    db_service.update_job_status(session, job, Status.in_progress)
```

Comment:
Wait, so we start the job IF we request the pipeline progress? It does not sound right. User could create a job, but did not start it on purpose. If FE requests the job status we must not start it, or modify it in any way. Just get the status and return it. Usually pending status is returned in such cases